src/Altinn.App.Core/Internal/Validation/ValidationService.cs

```diff
@@ -104,6 +104,33 @@ public async Task<List<ValidationIssue>> ValidateDataElement(Instance instance,
         return (await dataElementsIssuesTask).SelectMany(x => x).ToList();
     }
 
+    /// <inheritdoc/>
+    public async Task<List<ValidationIssue>> ValidateFileUpload(Instance instance, DataType dataType, byte[] fileContent, string? filename, string? mimeType, string? language)
+    {
+        ArgumentNullException.ThrowIfNull(instance);
+        ArgumentNullException.ThrowIfNull(dataType);
+        ArgumentNullException.ThrowIfNull(fileContent);
+
+        var validators = _validatorFactory.GetFileUploadValidators(dataType.Id);
+        var issuesLists = await Task.WhenAll(validators.Select(async v =>
```

Comment:
really nitpicking probably: `Task.WhenAll` leads to unbounded concurrency (potentially a lot of tasks are queued on the scheduler, though that is probably never the case in our context). Another option would be `Pallel.ForEachAsync`, where the default level of parallelism is equal the the logical processor count. This doesn't return anything though, so we would have to do the "gather"/"reduce" part on our own

Reply:
Is it really unbounded when it is limited by the number of validators the app developer adds.

I consciously chose this approach because validation is likely to be reasonably fast, unless it makes an external http call, in which case I don't think it makes sense to limit the parallelism.

Reply:
Used the word unbounded to indicate that _we_ don't enforce any bound. In terms of defensive programming I tend to try to put bounds on things so that even if users do something that isn't reasonable, the runtime of my code is still fairly predictable. But yeah it seems really really unlikely to ever be an issue here😄 

Yeah I agree that the "ideal bound" increases the more async the workload is,